### ISSUE TITLE
Bounce and Shake Effects

### ISSUE THREAD

**Issue by hansnolte** [MaurycyLiebner/enve#103]
Hello everyone,

since I don't want to set the keyframes by hand every time, I wanted to create a small library with different effects.
Something like this effects (under bounce and other effects)
http://www.theappguruz.com/tag-tools/web/CSSAnimations/

But how?
I'm afraid the only way is through expressions.

Is there an Enve-Expresssion tutorial or examples of this kind of effects?
Or is there another, simpler solution?

I am grateful for every tip.

Many greetings from a non-mathematician

**Comment by MaurycyLiebner** [comment]
Expressions are JavaScript-based. Bindings are the only non-JavaScript part. They are rather simple:
`variableName = object name.property group.property`. You can use bound values in the script portion of the expression.

**Comment by MaurycyLiebner** [comment]
I think the easier solution is to create the shake effect using keyframes and then copy/paste keyframes  between objects. You can also group object(s) and paste the keyframes to the group transformation, this way the shake effect will not interfere with existing object transformation keyframes.
![35](https://user-images.githubusercontent.com/16670651/79847205-aa74e080-83bf-11ea-8853-920fc238e054.png)

**Comment by MaurycyLiebner** [comment]
If you want the object to shake twice, you can select relevant keyframes and then duplicate them using **Ctrl + D** or **Shift + D**.

**Comment by hansnolte** [comment]
Hi Maurycy,

I actually wanted to avoid the copying and pasting of keyframes, since it is very time-consuming with many individual objects and scenes. 

Do you have an Expression example of a bounce effect?
Are After Effects Expression (partially) compatible with Enve Expression?

Many greetings
Norbert

**Comment by MaurycyLiebner** [comment]
No, there are no such examples as it would be more of a JavaScript/math example than an enve one. I used After Effects Expressions only few times, not extensively, and a long time ago.
Enve expressions are plain JavaScript (+ bindings), so I assume they are different.

Copying and pasting scripts will be even more time consuming.

**Comment by hansnolte** [comment]
Thanks for your answer.
Then I'll have to see how best to solve the problem.

**Comment by fodiator** [comment]
@hansnolte do you have already managed to create a solution?  I am struggling with this too

@MaurycyLiebner is there a forum established where enve users can share their knowledge, experience, questions and solutions?

**Comment by hansnolte** [comment]
> do you have already managed to create a solution?

I'm afraid not.
The only way is to set the keyframes by hand.

When I read that Maurycy is also a Blender user, this came to my mind immediately.
That would of course be the perfect solution.


![bounce](https://user-images.githubusercontent.com/7922627/82559702-5d2b9080-9b70-11ea-8112-0fae8a2f1609.gif)
